producers/pairquantities.py

```diff
@@ -716,3 +717,135 @@
         jet_hemisphere,
     ],
 )
+
+p4_fastmtt_mt = Producer(
+    name="p4_fastmtt_mt",
+    call='quantities::p4_fastmtt({df}, {output}, {input}, "mt")',
+    input=[
+        q.pt_1,
+        q.pt_2,
+        q.eta_1,
+        q.eta_2,
+        q.phi_1,
+        q.phi_2,
+        q.mass_1,
+        q.mass_2,
+        q.met,
+        q.metphi,
+        q.metcov00,
+        q.metcov01,
+        q.metcov11,
+        q.decaymode_2,
+        q.decaymode_2,
+    ],
+    output=[q.p4_fastmtt],
+    scopes=["mt"],
+)
+p4_fastmtt_et = Producer(
+    name="p4_fastmtt_et",
+    call='quantities::p4_fastmtt({df}, {output}, {input}, "et")',
+    input=[
+        q.pt_1,
+        q.pt_2,
+        q.eta_1,
+        q.eta_2,
+        q.phi_1,
+        q.phi_2,
+        q.mass_1,
+        q.mass_2,
+        q.met,
+        q.metphi,
+        q.metcov00,
+        q.metcov01,
+        q.metcov11,
+        q.decaymode_2,
+        q.decaymode_2,
+    ],
+    output=[q.p4_fastmtt],
+    scopes=["et"],
+)
+p4_fastmtt_tt = Producer(
+    name="p4_fastmtt_tt",
+    call='quantities::p4_fastmtt({df}, {output}, {input}, "tt")',
+    input=[
+        q.pt_1,
+        q.pt_2,
+        q.eta_1,
+        q.eta_2,
+        q.phi_1,
+        q.phi_2,
+        q.mass_1,
+        q.mass_2,
+        q.met,
+        q.metphi,
+        q.metcov00,
+        q.metcov01,
+        q.metcov11,
+        q.decaymode_2,
+        q.decaymode_2,
+    ],
+    output=[q.p4_fastmtt],
+    scopes=["tt"],
+)
+p4_fastmtt_em = Producer(
+    name="p4_fastmtt_em",
+    call='quantities::p4_fastmtt({df}, {output}, {input}, "em")',
+    input=[
+        q.pt_1,
+        q.pt_2,
+        q.eta_1,
+        q.eta_2,
+        q.phi_1,
+        q.phi_2,
+        q.mass_1,
+        q.mass_2,
+        q.met,
+        q.metphi,
+        q.metcov00,
+        q.metcov01,
+        q.metcov11,
+    ],
```

Comment:
`q.decaymode_*` missing, since it is an input for `quantities::p4_fastmtt()`.

Reply:
I fix those decaymodes in the implementation of the producer, not a very nice way, but it works

Reply:
I think that's fine.